src/main/java/Duke/tasks/Task.java

```diff
@@ -0,0 +1,108 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.ui.DukeException;
+/**
+ * Represents the task user want to do.
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor for Task.
+     *
+     * @param description description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns X if task is done, else empty.
+     *
+     * @return "X" if task is done, else " ".
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Returns the description of the task.
+     *
+     * @return description of the task.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Sets the task that is done to true.
+     *
+     */
+    public void setChecked() {
+        isDone = true;
+    }
+
+    /**
+     * Sets the task that is not done to false.
+     *
+     */
+    public void setUnchecked() {
+        isDone = false;
+    }
+
+    public abstract String saveToFileString();
```

Comment:
`saveToFileString` sounds like you are saving something inside the method, as the action verb is `save`. I would name this as `encodeTaskToString`.